### ISSUE TITLE
IllegalArgumentException: Cannot parse version '0.01.0' with '0Y.0M.MICRO'

### ISSUE THREAD

**Issue by p-marcin** [jreleaser/jreleaser#1728]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

I am trying to integrate jreleaser with my repo: https://github.com/p-marcin/java-dev-vm

With the following `jreleaser.yml`:
```yaml
project:
  versionPattern: CALVER:0Y.0M.MICRO

release:
  github:
    owner: p-marcin
    overwrite: true
    releaseName: '{{tagName}}'
    draft: true
    changelog:
      formatted: ALWAYS
      preset: conventional-commits
      format: '- {{commitShortHash}} {{commitTitle}}'
      categoryTitleFormat: '### {{categoryTitle}}'
      contributors:
        enabled: false
```

Run the command:
`jreleaser release -D=jreleaser.project.version=24.09.6 --dry-run --output-directory=target` 

Here is the output of my `git tag`:
```
v24.08.1
v24.08.2
v24.08.3
v24.09.1
v24.09.2
v24.09.3
v24.09.4
v24.09.5
```

### Expected Behaviour

Dry-run is completed without an exception.

### Actual Behaviour

Exception is thrown:
```
java.lang.IllegalArgumentException: Cannot parse version '0.01.0' with '0Y.0M.MICRO'
	at org.jreleaser.version.CalVer.of(CalVer.java:460)
	at org.jreleaser.version.CalVer.defaultOf(CalVer.java:466)
	at org.jreleaser.model.internal.util.VersionUtils.defaultVersion(VersionUtils.java:94)
	at org.jreleaser.sdk.git.ChangelogGenerator.defaultVersion(ChangelogGenerator.java:169)
	at org.jreleaser.sdk.git.ChangelogGenerator.resolveTags(ChangelogGenerator.java:209)
	at org.jreleaser.sdk.git.ChangelogGenerator.resolveCommits(ChangelogGenerator.java:308)
	at org.jreleaser.sdk.git.ChangelogGenerator.createChangelog(ChangelogGenerator.java:102)
	at org.jreleaser.sdk.git.ChangelogGenerator.generate(ChangelogGenerator.java:632)
	at org.jreleaser.sdk.git.ChangelogProvider.resolveChangelog(ChangelogProvider.java:121)
	at org.jreleaser.sdk.git.ChangelogProvider.getChangelog(ChangelogProvider.java:66)
	at org.jreleaser.sdk.github.GithubReleaser.generateReleaseNotes(GithubReleaser.java:104)
	at org.jreleaser.engine.changelog.Changelog.createChangelog(Changelog.java:38)
	at org.jreleaser.engine.release.ChangelogResolver.resolve(ChangelogResolver.java:53)
	at org.jreleaser.workflow.ChangelogWorkflowItem.doInvoke(ChangelogWorkflowItem.java:35)
	at org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at org.jreleaser.engine.hooks.HookExecutor.execute(HookExecutor.java:70)
	at org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at org.jreleaser.workflow.WorkflowImpl.doExecute(WorkflowImpl.java:129)
	at org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:54)
	at org.jreleaser.cli.Release.doExecute(Release.java:385)
	at org.jreleaser.cli.AbstractModelCommand.execute(AbstractModelCommand.java:90)
	at org.jreleaser.cli.Release.execute(Release.java:311)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:52)
	at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:37)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at org.jreleaser.cli.Main.execute(Main.java:98)
	at org.jreleaser.cli.Main.run(Main.java:81)
	at org.jreleaser.cli.Main.main(Main.java:70)
```

Attaching: [trace.log](https://github.com/user-attachments/files/17080231/trace.log)

I also get the same exception when using: `versionPattern: CALVER:YY.0M.MICRO`

### Environment Information

- **Operating System**: Ubuntu
- **JReleaser Version:** 1.14.0
- **JDK Version:** 21.0.4 (Oracle Corporation 21.0.4+8-LTS-274)

**Comment by P-Marcin** [comment]
I think the bug is here:
![obraz](https://github.com/user-attachments/assets/c8461743-cf8f-410e-a6b0-286d463caf57)
vs Pattern:
![obraz](https://github.com/user-attachments/assets/e3a547ee-a2ec-4c6b-9a02-8027bfa8fc6b)

**Comment by aalmiray** [comment]
I've get the same failure when calling the `of` method directly

```
Cannot parse version '0.01.0' with '0Y.0M.MICRO'
java.lang.IllegalArgumentException: Cannot parse version '0.01.0' with '0Y.0M.MICRO'
	at org.jreleaser.version.CalVer.of(CalVer.java:465)
	at org.jreleaser.version.CalVerTest.testVersionParsing(CalVerTest.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
```

**Comment by aalmiray** [comment]
The format `0Y.0M.MICRO` is expanded to the following pattern `^([1-9]|[1-9][0-9]|[1-9][0-9]{2})\.(0[1-9]|1[0-2])\.(0|[1-9]\d*)$` which clearly won't match a leading `0` in the `year` token.

**Comment by aalmiray** [comment]
Change must be done here https://github.com/jreleaser/jreleaser/blob/7e909958b12823fe5cd9427cfde889fa496ee21c/api/jreleaser-model-api/src/main/java/org/jreleaser/version/CalVer.java#L74 with pattern set to `"(0[0-9]|[1-9][0-9]|[1-9][0-9]{2})"`

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.15.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.15.0))
